chain/badtscache.go

```diff
@@ -21,10 +21,15 @@ func NewBadBlockCache() *BadBlockCache {
 	}
 }
 
-func (bts *BadBlockCache) Add(c cid.Cid) {
-	bts.badBlocks.Add(c, nil)
+func (bts *BadBlockCache) Add(c cid.Cid, reason string) {
```

Comment:
How about we use `error` for this, and `xerrors.Errorf("%w")` or `errors.Errorf("%w")` for chaining.

Reply:
The problem with chaining here is that it can be abused to OOM the node. Every block that links to a previous one would then contain the entirety of the previous error message, plus some. This would grow pretty quickly over time.